crates/starknet-devnet-core/src/messaging/mod.rs

```diff
@@ -82,20 +82,20 @@ impl MessagingBroker {
 
     /// Returns a reference to the ethereum instance if configured, an error otherwise.
     pub fn ethereum_ref(&self) -> DevnetResult<&EthereumMessaging> {
-        if self.ethereum.is_none() {
-            return Err(Error::MessagingError(MessagingError::NotConfigured));
+        if let Some(ethereum) = self.ethereum.as_ref() {
+            Ok(ethereum)
+        } else {
+            Err(Error::MessagingError(MessagingError::NotConfigured))
```

Comment:
Can be simplified as:
```suggestion
            self.ethereum.as_ref().ok_or(Error::MessagingError(MessagingError::NotConfigured))
```